src/view/layer-shell.cpp

```diff
@@ -467,6 +467,28 @@ void wayfire_layer_shell_view::unmap()
     wf_layer_shell_manager::get_instance().handle_unmap(this);
 }
 
+static bool same_state(const wlr_layer_surface_v1_state* a, const wlr_layer_surface_v1_state* b)
+{
+    if(a->committed == b->committed &&
+            a->anchor == b->anchor &&
+            a->exclusive_zone == b->exclusive_zone &&
+            a->margin.top == b->margin.top &&
+            a->margin.bottom == b->margin.bottom &&
+            a->margin.left == b->margin.left &&
+            a->margin.right == b->margin.right &&
+            a->keyboard_interactive == b->keyboard_interactive &&
+            a->desired_width == b->desired_width &&
+            a->desired_height == b->desired_height &&
+            a->layer == b->layer &&
+            a->actual_width == b->actual_width &&
```

Comment:
actual_width doesn't seem relevant, do we use it anywhere in this file? I think this is something wlroots uses for internal bookkeeping.

Reply:
I didn't check. All fields are exported so I compare all sensible ones to lock my screen quietly. I can remove it if it's not useful.

Reply:
On a second thought, we can also just look at the state->committed field. It will be nonzero iff anything of value changes, so we don't need to list all of this.

Reply:
Yes, that seems to work (tested in a vm).